run/authentication/auth_test.py

```diff
@@ -0,0 +1,79 @@
+# Copyright 2020 Google, LLC.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This test deploys a secure application running on Cloud Run
+# to test that the authentication sample works properly.
+
+import auth
+
+import os
+import subprocess
+from urllib import request
+import uuid
+
+import pytest
+
+
+@pytest.fixture()
+def service():
+    # Unique suffix to create distinct service names
+    suffix = uuid.uuid4().hex
+    project = os.environ['GOOGLE_CLOUD_PROJECT']
+
+    # Deploy a hello-world Cloud Run Service
+    subprocess.run(
+        [
+            "gcloud",
+            "run",
+            "deploy",
+            f"helloworld-{suffix}",
+            "--image",
+            "gcr.io/gcbdocs/hello",
+            "--platform=managed",
+            "--region=us-central1",
+            "--no-allow-unauthenticated",
+            "--quiet",
+        ], check=True
```

Comment:
Can you combine some of these?
I'd be more readable if we had something like:

```py
[
  f"gcloud run deploy helloworld{suffix}",
  "--image gcr.io/gcbdocs/hello",
  "--plaform=managed",
  ...
]
```

With groupings as appropriate.

Reply:
Unfortunately, we can only combine the flags and values -- all other parts of the command have to be comma separated.  But, I've updated to look a little more grouped:

```
"gcloud", "run", "deploy", f"helloworld-{suffix}",
"--image=gcr.io/gcbdocs/hello",
"--platform=managed",
"--region=us-central1",
"--no-allow-unauthenticated",
"--quiet"
```

wdyt?

Reply:
Great!